### PR TITLE
feat: [CDS-72170]: added secret manager field

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

after adding secret manager field: 

Screen Shot 2023-06-23 at 10 56 14 AM

### CODE REVIEW

**Comment by sahithikolichala** [comment]
retrigger cypress-cv

**Comment by sahithikolichala** [comment]
retrigger cypress-cv

**Comment by sahithikolichala** [comment]
retrigger cypress-cv

**Review by tan-nhu — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx [hunk 1]
```diff
@@ -872,8 +912,36 @@ export default function TerraformEditView(
...
+              {CDS_TERRAFORM_CLI_OPTIONS_NG &&
+              values?.spec?.configuration?.type === ConfigurationTypes.InheritFromApply ? (
+                <Accordion className={stepCss.accordion}>
+                  <Accordion.Panel
+                    id="step-1"
+                    summary={getString('cd.commandLineOptions')}
+                    details={
+                      <div className={css.optionalConfigDetails}>
+                        {initialValues?.type === StepType.TerraformDestroy &&
+                          values?.spec?.configuration?.type === ConfigurationTypes.InheritFromApply &&
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/Editview/TerraformEditView.tsx:928]: this condition can be removed as the accordion is conditionally rendered using the same condition

File: src/modules/75-cd/components/PipelineSteps/TerraformApply/__tests__/__snapshots__/TerraformApply.test.tsx.snap [hunk 2]
```diff
@@ -1487,6 +1487,9 @@ exports[`Test TerraformApply should render edit view as edit step - Inheritfromp
         </div>
       </div>
     </div>
+    <div
+      class="Accordion--accordion accordion"
+    />
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/TerraformApply/__tests__/__snapshots__/TerraformApply.test.tsx.snap:1492]: This snapshot is not useful. Please update the tests to enable the FF and to satisfy any conditions required to render the content inside the accordion.

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/TerraformInterfaces.ts [hunk 3]
```diff
@@ -721,38 +741,82 @@ export const getTerraformInitialValues = (data: any): TerraformData => {
...
+                      flag: commandFlag.flag
+                    })
+                  ),
+
+                  spec: { ...terraformSpec }
+                }
+              })
+        }
+      })
+    }
```
Comment (PratyushGar) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/TerraformInterfaces.ts:819]: nit:  
```suggestion
   
  return {
    ...data,
    ...(!isEmpty(path) && {
      spec: {
        ...data.spec,
        ...(path === 'spec.configuration'
          ? {
              configuration: {
                type: defaultTo(data.spec?.configuration?.type, ConfigurationTypes.Inline),
                skipRefreshCommand: data.spec?.configuration?.skipRefreshCommand,
                commandFlags: get(data, `${path}.commandFlags`)?.map(
                  (commandFlag: { commandType: string; flag: string }) => ({
                    commandType: commandFlag.commandType,
                    flag: commandFlag.flag
                  })
                ),
                ...(secretManagerField && {
                  encryptOutput: {
                    outputSecretManagerRef: secretManagerField
                  }
                }),
                spec: { ...terraformSpec }
              }
            }
          : {
              cloudCliConfiguration: {
                commandFlags: get(data, `${path}.commandFlags`)?.map(
                  (commandFlag: { commandType: string; flag: string }) => ({
                    commandType: commandFlag.commandType,
                    flag: commandFlag.flag
                  })
                ),
                ...(secretManagerField && {
                  encryptOutput: {
                    outputSecretManagerRef: secretManagerField
                  }
                }),
                spec: { ...terraformSpec }
              }
            })
      }
    })
  }
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terraform/TerraformInterfaces.ts [hunk 4]
```diff
@@ -721,38 +741,82 @@ export const getTerraformInitialValues = (data: any): TerraformData => {
...
+                    (commandFlag: { commandType: string; flag: string }) => ({
+                      commandType: commandFlag.commandType,
+                      flag: commandFlag.flag
+                    })
+                  ),
+                  encryptOutput: {
+                    outputSecretManagerRef: secretManagerField
+                  },
+                  spec: { ...terraformSpec }
+                }
```
Comment (PratyushGar) [inline src/modules/75-cd/components/PipelineSteps/Common/Terraform/TerraformInterfaces.ts:765]: nit:  This if else block can be removed using conditionally spread for `encryptOutput`
```suggestion
   
  return {
    ...data,
    ...(!isEmpty(path) && {
      spec: {
        ...data.spec,
        ...(path === 'spec.configuration'
          ? {
              configuration: {
                type: defaultTo(data.spec?.configuration?.type, ConfigurationTypes.Inline),
                skipRefreshCommand: data.spec?.configuration?.skipRefreshCommand,
                commandFlags: get(data, `${path}.commandFlags`)?.map(
                  (commandFlag: { commandType: string; flag: string }) => ({
                    commandType: commandFlag.commandType,
                    flag: commandFlag.flag
                  })
                ),
                ...(secretManagerField && {
                  encryptOutput: {
                    outputSecretManagerRef: secretManagerField
                  }
                }),
                spec: { ...terraformSpec }
              }
            }
          : {
              cloudCliConfiguration: {
                commandFlags: get(data, `${path}.commandFlags`)?.map(
                  (commandFlag: { commandType: string; flag: string }) => ({
                    commandType: commandFlag.commandType,
                    flag: commandFlag.flag
                  })
                ),
                ...(secretManagerField && {
                  encryptOutput: {
                    outputSecretManagerRef: secretManagerField
                  }
                }),
                spec: { ...terraformSpec }
              }
            })
      }
    })
  }
```

**Review by tan-nhu — DISMISSED** [review]

**Review by PratyushGar — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

**Comment by sahithikolichala** [comment]
retrigger cypress-pipeline

**Review by rameshx — APPROVED** [review]

**Pull request merged**